tests/unit/migrator/test_migrator_api.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+from mock import MagicMock, patch
+
+from inspirehep.migrator.api import continuous_migration
+
+
+@patch("inspirehep.migrator.api.current_app")
+@patch("inspirehep.migrator.api.StrictRedis.from_url")
+@patch("inspirehep.migrator.api.Lock")
+@patch("inspirehep.migrator.api.insert_into_mirror")
+@patch("inspirehep.migrator.api.zlib")
+@patch("inspirehep.migrator.api.migrate_from_mirror")
+@patch("inspirehep.migrator.api.wait_for_all_tasks")
+def test_continous_migration_when_last_record_migrated_wont_be_added_to_mirror_properly(
+    wait_for_all_tasks_mock,
+    migrate_from_mirror_mock,
+    zlib_mock,
+    insert_into_mirror_mock,
+    lock_mock,
+    strict_redis_mock,
+    current_app_mock,
+):
+    strict_redis_mock().llen.side_effect = [2, 2, 1, 0]
+    strict_redis_mock().lrange.side_effect = [[True], [True], [False]]
+    lock_mock.acquire.side_effect = True
+    insert_into_mirror_mock.side_effect = [[1, 2, 3], []]
+
+    continuous_migration()
+    migrate_from_mirror_mock.assert_called_once()
```

Comment:
:/ to be honest, I don't understand this test. The side effects like ``[2, 2, 1, 0]`` what does it mean? Why it's not ``[2, 1, 0]``? And ``insert_to_mirror`` why returns ```[1,2,3]```? 

Why we don't write a proper integration test and check that both records are there?

Reply:
This is simple logic of continuous_migration function. Its perfect for unit test and not for integration test. Here you do not want to check any data, or interaction between modules. You only want to check what will happen if last response from `insert_into_mirror` is empty list. Side effects are also easy to understand if you look to continuous_migration function.

Reply:
Honestly, for me at least are not easy to understand, I have to go and check each individual function and see what it expects otherwise I'm lost. 

Also I think there is an issue with:
```strict_redis_mock().llen.side_effect = [2, 2, 1, 0]```
should be 
strict_redis_mock().llen.side_effect = [2, 1, 0]

Reply:
No it shouldn't, as there is first call for `redis_cli.llen()` which is providing value for logger.info message (which happens only once). Second and latter calls are for while loop. I can add more comments in this test.

Reply:
Thanks for adding the comments. IMO adding commends to explain 5 lines of tests there is a problem. Maybe someone else can commend if she/he understands what is going on